File: week6/DictionaryAttack.rb
# !/usr/bin/ruby 
# Author: Content provided in course work
# Created 18/08/2021 8pm



       require 'digest' 
      
       #read the password hash 
    
       passwordHash = "61ea0803f8853523b777d414ace3130cd4d3f92de2cd7ff8695c337d79c2eeee" 
             
       #open the wordlist and repeat for each word 
             
       IO.foreach("lg_word.txt") do | word | 
             
           word.chomp! 
             
           #hash the word 
             
           wordlistHash = Digest::SHA256.hexdigest(word) 
             
           puts("Trying password #{word}: #{wordlistHash}") 
             
           #if the hash is the same as the correct password's hash then we have cracked the password! 
             
           if(wordlistHash == passwordHash) 
             
               puts("Password has been cracked! It was #{word}") 
             
               return 
             
           end 
             
       end 
            

File: week6/DictionaryAttack.py
# !/usr/bin/python3 
# Created: 18/08/2021
# Author: Provided by course work      

# Start

import hashlib
      
# hidden password hash 
      
passwordHash = "5994471abb01112afcc18159f6cc74b4f511b99806da59b3caf5a9c173cacfc5" 
      
# open the wordlist 
      
with open("lg_word.txt", "r") as wordlist: 
      
    # repeat for each word 
      
    for word in wordlist.readlines(): 
      
        word = word.rstrip() 
      
        # hash the word 
      
        wordlistHash = hashlib.sha256(word.encode("utf-8")).hexdigest() 
        print("Trying password" ,word ,wordlistHash)



 #       print(f"Trying password {word}:{wordlistHash}") 
      
        # if the hash is the same as the correct password's hash then we have cracked the password! 
      
        if(wordlistHash == passwordHash): 
      
            print("Password has been cracked! It was " ,word) 
      
            break 

# End  